embed/etcd.go

```diff
@@ -248,32 +255,46 @@ func (e *Etcd) Close() {
 	// close all idle connections in peer handler (wait up to 1-second)
 	for i := range e.Peers {
 		if e.Peers[i] != nil && e.Peers[i].close != nil {
-			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
-			e.Peers[i].close(ctx)
-			cancel()
+			e.Peers[i].close(time.Second)
 		}
 	}
 }
 
-func (e *Etcd) stopGRPCServer(gs *grpc.Server) {
-	timeout := 2 * time.Second
-	if e.Server != nil {
-		timeout = e.Server.Cfg.ReqTimeout()
+func stopServers(ss *servers, timeout time.Duration) {
```

Comment:
why not take a context but time.duration?

Reply:
Ok, that works too. Let me change to context.